src/Validators/TokenValidator.php

```diff
@@ -36,10 +36,18 @@ public function check($value)
      */
     protected function validateStructure($token)
     {
-        if (count(explode('.', $token)) !== 3) {
+        $parts = explode('.', $token);
+
+        if (count($parts) !== 3) {
             throw new TokenInvalidException('Wrong number of segments');
         }
 
+        $parts = array_filter(array_map('trim', $parts));
+
+        if (count($parts) !== 3 || implode('.', $parts) !== $token) {
```

Comment:
Nope because, in this case we test the filtered array.

First test: `one..three` => 3 parts
Second test: `one.three` => 2 parts